File: scratch.py
import urllib.request
from bs4 import BeautifulSoup
import re

req = urllib.request.Request('https://www.abc.es/espana/abci-sanchez-viaja-marriuecos-crisis-202302011234_noticia.html', headers={'User-Agent': 'Mozilla/5.0'})
try:
    html = urllib.request.urlopen(req).read().decode('utf-8')
except Exception as e:
    req = urllib.request.Request('https://www.abc.es/', headers={'User-Agent': 'Mozilla/5.0'})
    html = urllib.request.urlopen(req).read().decode('utf-8')
    links = re.findall(r'href="(https://www.abc.es/.*?\.html)"', html)
    url = links[0]
    html = urllib.request.urlopen(urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})).read().decode('utf-8')

soup = BeautifulSoup(html, 'html.parser')
scripts = soup.find_all('script')
for s in scripts:
    if s.get('id') == 'evo-swg-markup':
        print("Found evo-swg-markup")
        print(s.string[:500])
        print("-------")
        print(s.string[-1000:])
        break
else:
    print("No evo-swg-markup found")
    # print body paragraphs
    ps = soup.find_all('p')
    for p in ps:
        if len(p.text) > 100:
            print(p.text[:100])

File: scratch.php
<?php
$url = 'https://www.abc.es/espana/madrid/abci-ayuso-carga-contra-sanchez-202302011234_noticia.html'; // Let's fetch one we know exists or find one
$html = file_get_contents('https://www.abc.es/sociedad/aemet-avisa-espana-sobre-tiempo-hara-dia-20260811083846-nt.html');
$dom = new DOMDocument();
@$dom->loadHTML($html);
$xpath = new DOMXPath($dom);
$scriptNodes = $xpath->query('//script[@id="evo-swg-markup"]');
if ($scriptNodes->length > 0) {
    $json = $scriptNodes->item(0)->textContent;
    $data = json_decode($json, true);
    $body = $data['articleBody'] ?? '';
    echo substr($body, 0, 500) . "\n\n-----\n\n";
    echo substr($body, -1000);
} else {
    echo "No schema found";
}
